File: AstroDroid.Core/Validators/MessageValidator.cs
using AstroDroid.Core.Interfaces;
using FluentValidation;

namespace AstroDroid.Core.Validators
{
    /// <summary>
    /// Validates node messages
    /// </summary>
    public class MessageValidator : AbstractValidator<INodeMessage>
    {
        public MessageValidator()
        {
            RuleFor(x => x.Sender).NotNull().NotEmpty();
            RuleFor(x => x.Topic).NotNull().NotEmpty();
            RuleFor(x => x.Type).NotNull().NotEmpty();
        }
    }
}

File: AstroDroidJs.Core/src/validators/message-validator.ts

import { INodeMessage } from "../interfaces/node-message";
import {AbstractValidator, Severity} from "fluent-ts-validator";

export class MessageValidator extends AbstractValidator<INodeMessage> {
  constructor() {
    super();
 
    this.validateIfString(message => message.Content).isNotNull().isNotEmpty();
    this.validateIfString(message => message.Sender).isNotNull().isNotEmpty();
    this.validateIfString(message => message.Topic).isNotNull().isNotEmpty();
    this.validateIfString(message => message.Type).isNotNull().isNotEmpty();
  }
}